src/java.base/share/classes/com/sun/crypto/provider/GCTR.java

```diff
@@ -228,15 +228,19 @@ public int update(ByteBuffer src, ByteBuffer dst) {
         len = src.remaining() - (src.remaining() % blockSize);
         int processed = len;
         byte[] in = new byte[Math.min(MAX_LEN, len)];
-        while (processed > MAX_LEN) {
-            src.get(in, 0, MAX_LEN);
-            encrypt(in, 0, MAX_LEN, in, 0);
-            dst.put(in, 0, MAX_LEN);
-            processed -= MAX_LEN;
+        try {
+            while (processed > MAX_LEN) {
+                src.get(in, 0, MAX_LEN);
+                encrypt(in, 0, MAX_LEN, in, 0);
+                dst.put(in, 0, MAX_LEN);
+                processed -= MAX_LEN;
+            }
+            src.get(in, 0, processed);
+            encrypt(in, 0, processed, in, 0);
+            dst.put(in, 0, processed);
+        } finally {
+            Arrays.fill(in, (byte)0);
```

Comment:
The same handling should be applied to the 'out' buffer allocated on line 177 assuming the idea is to not leaving copies of user data lying around on the heap. Recovered text may contain the same bytes as plain text, so arguably the same handling applied to both input/output?

Reply:
Sounds good. Rest of changes look fine.